parallel-consumer-python/tests/load/test_java_clients.py

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright (C) 2020-2022 Confluent, Inc.
+#
+
+from busypie import wait, MINUTE
+from jpype import JProxy
+
+from pyallel_consumer import start_jvm
+from tests import timeit, sleep_a_bit
+
+start_jvm()
+
+from java.util.concurrent import Callable
+from java.util.concurrent.atomic import AtomicInteger
+from java.util.concurrent.Executors import newCachedThreadPool
+
+from org.apache.kafka.clients.producer import KafkaProducer
+from org.apache.kafka.clients.consumer import KafkaConsumer
+
+from jio.confluent.parallelconsumer import ParallelConsumerOptions, ParallelStreamProcessor
+
+MAX_TIMEOUT = (15, MINUTE)
+
+
+@timeit
+def consume_messages(consumer: KafkaConsumer, num_messages: int):
+    count = AtomicInteger(0)
+    all_records = []
+
+    def process_messages():
+        nonlocal count
+        while count.get() < num_messages:
+            records = list(consumer.poll(500).iterator())
+            print(f'Polling batch of {len(records)} messages')
+            for _ in records:
+                sleep_a_bit()
+            count.getAndAdd(len(records))
+            all_records.extend(records)
+
+    newCachedThreadPool().submit(JProxy(Callable, dict(call=process_messages)))
+    success_criterion = lambda: count.get() == num_messages
+    wait().at_most(*MAX_TIMEOUT).until(success_criterion)
+    assert success_criterion()
+
+
+def test_normal_kafka_consumer(loaded_topic: str, consumer: KafkaConsumer, num_messages: int):
+    consumer.subscribe([loaded_topic])
+    print('Starting to read back')
+    consume_messages(consumer, num_messages)
+
+
+@timeit
+def parallel_consume_messages(processor, num_messages):
+    count = AtomicInteger(0)
+
+    def process_message(message):
+        nonlocal count
+        sleep_a_bit()
+        count.getAndIncrement()
+
+    processor.poll(process_message)
+    success_criterion = lambda: count.get() >= num_messages
+    wait().at_most(*MAX_TIMEOUT).until(success_criterion)
+    assert success_criterion()
+
+
+def test_async_consume_and_process(loaded_topic: str, consumer: KafkaConsumer, transactional_producer: KafkaProducer,
```

Comment:
@astubbs it would be great to have a load test which is more representative of a use case where parallel consumer shines if you have advice on how to set this up
This is a port of the Java integration test [LoadTest](https://github.com/confluentinc/parallel-consumer/blob/master/parallel-consumer-core/src/test-integration/java/io/confluent/parallelconsumer/integrationTests/LoadTest.java)

Aka loading a topic with 4k messages and consuming with sleeps to represent database accesses
The performance results are:
- Normal Java Kafka Client accessed via Python: 0.43 seconds
- Parallel Consumer accessed via Python: 2.41 seconds
- Normal Python Kafka Client: 0.02 seconds

So in this use case parallel consumer is the worst option

Reply:
You can't do what PC does here, with the normal clients. PC enforces ordering by key, the other clients do not. What happens when you change line 70 from:

        .ordering(ParallelConsumerOptions.ProcessingOrder.KEY)


to:

        .ordering(ParallelConsumerOptions.ProcessingOrder.UNORDERED)

?

Reply:
That run, with `UNORDERED` I got

- Normal Java Kafka Client accessed via Python: 0.72 seconds
- Parallel Consumer accessed via Python: 2.70 seconds
- Normal Python Kafka Client: 0.16 seconds

So large variability in the others, but PC via Python in the same ballpark as before

Reply:
Changing a setting in PC, isn't going to affect the timing of the tests not using PC. Ping me on slack, and lets go through this together instead of like this.

Reply:
oh, I just noticed. Are the other two tests also using transactions? PC is also set to use transactions in the python test.
I see it's concurrency is only set to 3 as well - try 100, or 500.

Reply:
BTW, the test I was using in the weekend to test out the new engine speeds is `MultiInstanceHighVolumeTest`.

Reply:
Also, try it with the larger volumes of messages. Like several million.

Reply:
The async commit mode is probably what you want, unless the other tests are explicitly stopping to commit.

Reply:
And - try comparing it to using vanilla pc.

Reply:
Yeh there's too many settings to tweak that I don't know enough about
Will ping you :)